protocol/messenger.go

```diff
@@ -2485,49 +2488,66 @@ func (m *Messenger) DeleteAccount(address types.Address) error {
 	return m.syncWallets(accs, m.dispatchMessage)
 }
 
-// syncWallets syncs all wallets with paired devices
-func (m *Messenger) syncWallets(accs []*accounts.Account, rawMessageHandler RawMessageHandler) error {
-	if !m.hasPairedDevices() {
-		return nil
-	}
-
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	defer cancel()
-
-	clock, chat := m.getLastClockWithRelatedChat()
-
+func (m *Messenger) prepareSyncWalletAccountsMessage(accs []*accounts.Account, clock uint64) *protobuf.SyncWalletAccounts {
 	accountMessages := make([]*protobuf.SyncWalletAccount, 0)
 	for _, acc := range accs {
-		if acc.Type != accounts.AccountTypeWatch &&
-			acc.Type != accounts.AccountTypeGenerated {
+		if acc.Chat {
 			continue
 		}
+
+		// Once mobile app supports seed phrase and private key imported accounts we should remove the following `if` block
+		if isMobileApp {
+			if acc.Type != accounts.AccountTypeWatch &&
+				acc.Type != accounts.AccountTypeGenerated {
+				continue
+			}
+		}
 		var accountClock uint64
 		if acc.Clock == 0 {
```

Comment:
I am not sure we need this, if it was never modified, a clock of `0` should be fine (that's generally what we do, we only set a clock when things changes, if it's a default thing, the clock should be 0)
I think this is the same point as @qfrank 

Basically:

1) If something is defaulted, the clock should be set to 0
2) If the user makes an explicit action, the clock should be set to the time that decision was made

The code above, does not follow the rules above, since it looks like something might be defaulted (i.e has a clock of 0), but then the clock is set **at sync time**, which is not a pattern we generally use, and looks risky, since it might override an actual decision of the user on a different device.
Though I might be missing context.

Reply:
If we change it here (when messages are dispatched), then we need to change it on the handling side [here](https://github.com/status-im/status-go/blob/1fe4a898b60694077c6df6fef352c71bb9b7d8eb/protocol/messenger_handler.go#L2779) (if it remains 0 it won't be applied with that condition then).

@cammellos that's the code which was already here, not written by me and I didn't want to change it cause I still do think that we don't need to change it and the reasons:
- we do perform a backup every 8hrs and if we would change that it means when user recovers Status account we will overwrite each wallet account as many times as many backed up messages are received instead just discard older messages and apply only newer ones
- hypothetically :) - if user has 2 devices (with the same seed phrase) which are not synced, device A has wallet account named "AccA", and device B has the same wallet account named "AccB" and each device does backup every 8hrs. Then device A is lost, and user decide to delete account on device B and recover it. Then while recovering if device B firstly receives backed up message from device B and set the wallet account name to "AccB", then receives backed up message from device A and overwrite that account cause both have clock set to 0, the end result will be that wallet account on device B has name "AccA", which is not what user would expect.

Reply:
> If we change it here (when messages are dispatched), then we need to change it on the handling side [here](https://github.com/status-im/status-go/blob/1fe4a898b60694077c6df6fef352c71bb9b7d8eb/protocol/messenger_handler.go#L2779) (if it remains 0 it won't be applied with that condition then).

Just to make sure we are on the same page, I am not arguing for changing the clock on dispatch, the opposite, clock should never changed when dispatching sync messages, it should only changed in response of a user decision. (i.e we should always use `acc.Clock`, never chat.nextClock, if clock is 0, there's an argument for not syncing that piece of data, but that's dependent on what we are syncing)

The code above though, sets the clock to `chat.nextClock` if the account.Clock is == 0, which means it is setting the clock at dispatch time, but that's something we generally don't do as I can't think of a case where that's necessary (but maybe I am missing something).

Are we on the same page on this point? (from your reply it looked to me that you think I am arguing for changing clock value at dispatch time)

Reply:
> which means it is setting the clock at dispatch time

We don't set it to db for an account, but we set that `chat.nextClock` as `Clock` value only to dispatched message (one we're backing up) for that account.

How I understood you is that you're saying that we **_don't need_** to set `Clock` of `protobuf.SyncWalletAccount` to `chat.nextClock`, in case the db clock value is 0, at the moment when we're dispatching a message, but use `Clock` value which is set in the db for the account we're building a `protobuf.SyncWalletAccount` message.

Now what I am saying is, if we would set always a `Clock` value of `protobuf.SyncWalletAccount` to clock value from the `db`, then we would have many `protobuf.SyncWalletAccount` messages with the same `Clock` value which is `0`. That means at the moment user is doing an account recovering the app will overwrite an account in the db for each received backed up message, cause they will all have `0` value for `Clock`. (Points 1 and 2 from my previous comment).

Reply:
Ok, a question, when would an account have a clock value of 0?

Reply:
As I see in the desktop app accounts have always `0` value for clock untill one of the following is true:
- the app is recovered from backed up messages
- the app is synced with another app (happens when user is recovering Status account from other device)
- the app is paired with another app and account is updated on that other app

Reply:
if we update the account locally, it should set the clock value right?

the cases you mentioned are only in response to a network event, but `clock` must be set locally at some point right?

Reply:
> the cases you mentioned are only in response to a network event, but clock must be set locally at some point right?

That's not the case at the moment and has never been in the past as I see 🙂. When an account is updated locally the desktop app sets only `updated_at`, but that's because of [this line](https://github.com/status-im/status-go/blob/0eff61c57ae8fa1e80a3032e1ba8ada583598747/multiaccounts/accounts/database.go#L406)

Reply:
Though doesn't status-go needs to care about that, not a client?

Reply:
mmh, looks like then there's a bit of an issue :) 

Anytime the user makes a decision about a piece of data, it should update the clock, I am not sure about the lifecycle of accounts, but if we can update them (i.e change name/emojii), we should set the clock to the time it has been updated, so that changes are correctly propagated to clients

Reply:
> Though doesn't status-go needs to care about that, not a client?

correct, status-go should set the clock, `saveAccounts` though is a bit of an annoying endpoint, since we don't really now what has changed, it would depend on the usage I think

Reply:
I fully agree with all you've said in the last 2 messages and going to update that now.

Then I guess it does make sense to set `Clock` for newly created accounts as well? And then we can remove the discutable check if db clock is `0` or not, where we started this discussion from.

Reply:
> Then I guess it does make sense to set Clock for newly created accounts as well? And then we can remove the discutable check if db clock is 0 or not, where we started this discussion from.

It would generally make sense, yes, unless it's a default account generated without user input (i.e no need to backup/sync)

Reply:
@cammellos related changes are applied in [this commit](https://github.com/status-im/status-go/pull/3416/commits/dfa3cbbac0aed4ce2aee7e3779b3baeb6e11274a).

- setting clock when adding/updating accounts
- syncing/backing up accounts with the time they actually were updated instead the time when dispatching is done as it was before